### PR TITLE
refactor(security): Renames ServerGroupNameable to ServerGroupsNameab…

### PR DESCRIPTION
**Pull request by ttomsu**
Repurposes `ServerGroupNameable` to be singular equivalent of `ServerGroupsNameable`

### CODE REVIEW

**Comment by ttomsu** [comment]
https://github.com/ttomsu/clouddriver/pull/3 to be merged on top of this PR

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 Google, Inc.
```
Comment (jtk54) [inline clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java]: 2018

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupNameable.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016 Google, Inc.
+ * Copyright 2018 Google, Inc.
```
Comment (jtk54) [inline clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupNameable.java]: Generally we don't update this after file creation.

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import com.netflix.frigga.Names;
+
+import java.util.Collection;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Convenience trait for parsing application names out of a description with one or more server group names.
+ */
+public interface ServerGroupsNameable extends ApplicationNameable {
```
Comment (jtk54) [inline clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java]: What's the purpose of having both ServerGroup and ServerGroupsNameable?

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java (hunk 1 above)
Comment (ttomsu) [reply clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java]: See your other comment - we don't usually update this after file creation - it was just moved.

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupNameable.java (hunk 2 above)
Comment (ttomsu) [reply clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupNameable.java]: It's a new file

Comment (jtk54) [reply clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupNameable.java]: OK, the github diff is genuinely horrible and I can't tell what's going on.

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java (hunk 3 above)
Comment (ttomsu) [reply clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java]: Some operation descriptions declare fields/properties named `serverGroupName` vs the interface requiring `getServerGroupNames` - if the interface requires the plural version, it's name should reflect so (hence this PR)

Comment (jtk54) [reply clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupsNameable.java]: :+1: This was more confusion from github's rendering I think.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/RegisterInstancesWithGoogleLoadBalancerDescription.groovy [hunk 4]
```diff
@@ -16,9 +16,23 @@
...
+
+class RegisterInstancesWithGoogleLoadBalancerDescription extends AbstractGoogleCredentialsDescription implements ApplicationNameable{
   List<String> loadBalancerNames
   List<String> instanceIds
   String region
   String accountName
+
+  @Override
+  Collection<String> getApplications() {
+    def list = (loadBalancerNames - null)
```
Comment (ajordens) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/RegisterInstancesWithGoogleLoadBalancerDescription.groovy]: fwiw the `DescriptionAuthorizer` should handle nulls but it's no doubt safer to be explicit

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/TerminateGoogleInstancesDescription.groovy [hunk 5]
```diff
@@ -16,7 +16,9 @@
 
 package com.netflix.spinnaker.clouddriver.google.deploy.description
 
-class TerminateGoogleInstancesDescription extends AbstractGoogleCredentialsDescription {
+import com.netflix.spinnaker.clouddriver.security.resources.ServerGroupNameable
+
+class TerminateGoogleInstancesDescription extends AbstractGoogleCredentialsDescription implements ServerGroupNameable {
   String serverGroupName
   List<String> instanceIds
```
Comment (ajordens) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/TerminateGoogleInstancesDescription.groovy]: I don't know if you're checking the actual operation itself that the instanceIds belong to the specified server group ... in aws-land we weren't and pretty sure ended up fetching the instanceIds and grabbing the server group names off of them.

File: clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupNameable.java (hunk 2 above)
Comment (ajordens) [reply clouddriver-security/src/main/groovy/com/netflix/spinnaker/clouddriver/security/resources/ServerGroupNameable.java]: s/horrible/the best

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/TerminateGoogleInstancesDescription.groovy (hunk 5 above)
Comment (ttomsu) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/TerminateGoogleInstancesDescription.groovy]: I believe GCE instance IDs are unique numbers, and thus the only thing I _could_check is the name.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/RegisterInstancesWithGoogleLoadBalancerDescription.groovy (hunk 4 above)
Comment (ttomsu) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/RegisterInstancesWithGoogleLoadBalancerDescription.groovy]: Ack.

**Pull request merged**